holidays/countries/fiji.py

```diff
@@ -0,0 +1,188 @@
+    # %s (observed).
+    observed_label = "%s (observed)"
+    # %s (observed, estimated).
+    observed_estimated_label = "%s (observed, estimated)"
+    # Act No. 13 of 2015
+    start_year = 2016
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=FijiHinduHolidays, show_estimated=True)
+        IslamicHolidays.__init__(self, cls=FijiIslamicHolidays, show_estimated=True)
+        StaticHolidays.__init__(self, FijiStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday("Good Friday")
+
+        # Easter Saturday.
+        self._add_holy_saturday("Easter Saturday")
+
+        # Easter Monday.
+        self._add_easter_monday("Easter Monday")
+
+        # National Sports Day.
+        name = "National Sports Day"
+        if self._year == 2018:
+            self._add_holiday_jun_29(name)
+        elif self._year <= 2017:
+            self._add_holiday_jun_26(name)
+
+        if self._year <= 2022:
+            # Constitution Day.
+            self._add_holiday_sep_7("Constitution Day")
+
+        if self._year >= 2023:
+            # Girmit Day.
+            self._move_holiday(
+                self._add_holiday_may_14("Girmit Day"),
+                rule=ALL_TO_NEAREST_MON,
+                show_observed_label=False,
+            )
+
+        # Ratu Sir Lala Sukuna Day.
+        name = "Ratu Sir Lala Sukuna Day"
+        if self._year == 2023:
+            self._add_holiday_last_mon_of_may(name)
+        elif self._year >= 2024:
+            self._add_holiday_last_fri_of_may(name)
+
+        # Fiji Day.
+        self._add_holiday_oct_10("Fiji Day")
+
+        # Prophet Mohammed's Birthday
+        self._populate_observed(self._add_mawlid_day("Prophet Mohammed's Birthday"))
+
+        # Diwali
+        self._add_observed(self._add_diwali("Diwali"))
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
+
+        # Boxing Day.
+        self._add_observed(self._add_christmas_day_two("Boxing Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
+
+    def _populate_workday_holidays(self):
+        # Easter Sunday.
+        self._add_easter_sunday("Easter Sunday")
+
+        # Eid al-Fitr.
+        self._add_eid_al_fitr_day("Eid al-Fitr")
+
+        if self._year >= 2023:
+            # Constitution Day.
+            self._add_holiday_sep_7("Constitution Day")
+
+
+class FJ(Fiji):
+    pass
+
+
+class FJI(Fiji):
+    pass
+
+
+class FijiHinduHolidays(_CustomHinduHolidays):
+    # https://www.timeanddate.com/holidays/fiji/diwali
+    DIWALI_DATES = {
+        2016: (OCT, 27),
+        2017: (OCT, 16),
```

Comment:
```suggestion
        2016: (OCT, 31),
        2017: (OCT, 19),
```
Perhaps in 2016 the exact date is OCT 30, and OCT 31 is observed holiday (official announcements don't distinguish between them).